airflow/serialization/serialized_objects.py

```diff
@@ -127,7 +127,10 @@ def _get_registered_timetable(importable_string: str) -> Optional[Type[Timetable
     if importable_string.startswith("airflow.timetables."):
         return import_string(importable_string)
     plugins_manager.initialize_timetables_plugins()
-    return plugins_manager.timetable_classes.get(importable_string)
+    if plugins_manager.timetable_classes:
+        return plugins_manager.timetable_classes.get(importable_string)
+    else:
+        return None
```

Comment:
We should change these global variables to remove the `Optional` case.

Reply:
Did look into this in detail, those global variables (plugins_manager.timetable_classes etc.) are defined for plugins and if they are not provided/loaded/exists then they would be None. 
In general, these initializers (initialize_timetables_plugins(), integrate_executor_plugins() etc.) could be refactored to do get_something rather than using global variables but that's a bit larger change.